src/data/DataManager.js

```diff
@@ -15,45 +15,49 @@ export default class DataManager extends Component {
             loading: true,
             data: {}
         }
-    } 
+    }
 
-    loadData(paths) {
+    loadData(paths, param = "") {
         let dataLoad = []
         for (var index in paths) {
-            dataLoad.push(dataSource.GetData(paths[index]))
+            paths[index] == 'user' ? dataLoad.push(dataSource.GetData(paths[index], param)) : dataLoad.push(dataSource.GetData(paths[index]))
+            // dataLoad.push(dataSource.GetData(paths[index], param))
         }
         Promise.all(dataLoad).then((responses) => {
             let paths = DataMap[this.props.component]
             let data = responses.reduce((obj, res, i) => {
                 obj[paths[i]] = res.data
                 return obj
             }, {})
-            this.setState({ 
+            this.setState({
                 data: data,
                 loading: false
-             })
+            })
         })
     }
 
     selectComponent(comp) {
-        const wrap = (Component) =><Component {...this.state.data} />
+        const wrap = (Component) => <Component {...this.state.data} />
         switch (comp) {
-            case "AccountProfile" :
+            case "AccountProfile":
                 return wrap(AccountProfile)
-            case "JobProfile" :
+            case "JobProfile":
                 return wrap(JobProfile)
-            default : 
+            default:
                 return wrap(HomePage)
         }
     }
 
+    componentDidMount = () => {
+        let param = window.location.pathname.substr(16, 25)
```

Comment:
This substring seems dependent on this one endpoint. Can we pass the userId param from the router in as props of the DataManager?